src/coreclr/System.Private.CoreLib/src/System/Math.CoreCLR.cs

```diff
@@ -95,6 +95,11 @@ public static partial class Math
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe (double Sin, double Cos) SinCos(double x)
         {
+            if (RuntimeHelpers.IsKnownConstant(x))
+            {
+                return (Sin(x), Cos(x));
+            }
```

Comment:
Does this mean Sin(x) and Cos(x) already special-case known constants and that's why this helps SinCos?

Reply:
Correct and adding the JIT support for `SinCos` directly isn't worth the complexity. It would require us to plumb through support for value tuples and/or `out` parameters in various phases, which is a lot of work compared to this handling.